automerge-frontend/src/lib.rs

```diff
@@ -379,18 +385,18 @@ impl Frontend {
         }
     }
 
-    pub fn change<F, E>(
+    pub fn change<F, O, E>(
         &mut self,
         message: Option<String>,
         change_closure: F,
-    ) -> Result<Option<UncompressedChange>, E>
+    ) -> Result<(O, Option<UncompressedChange>), E>
```

Comment:
We could add a type here instead of the tuple, something like `ChangeResult`. I'm not sure if it would really clarify anything though, the existing types probably communicate intent clearly enough as `O` is obviously the return type of `F`.

I also think the number of type parameters on `change` is becoming really unpleasant to work with, but I think solving this is probably a matter of building richer APIs on top of `change` (e.g the `automergeable` work).

Reply:
I think it still makes sense to keep these values separate here, mainly since they serve two purposes, one for the user to do as they wish with and the other to be passed to a backend.